docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+can only be evaluated on a fixed grid, which is defined on initialization by the
+exposure map. In addition it takes reduced PSF and EDISP objects. In functionality
+it corresponds to the current `MapEvaluator`, but with the model parameters attached
+and the background handling removed. Additional "hidden" IRF parameters e.g. to propagate
+systematics could be optionally introduced later. This class will replace the current
+`MapEvaluator`.
+
+.. code-block::
+
+    from gammapy.cube import SourceIRFModel, SourceModel
+
+    model = SourceModel()
+    source_irf_model = SourceIRFModel(model, exposure=exposure, edisp=edisp, psf=psf)
+    npred = source_irf_model.npred
+    dnde = source_irf_model.dnde
+    flux = source_irf_model.flux
+
+
+Improve `SourceModels` class
+----------------------------
+The existing `CompoundSkyModel` is a nice very generic abstraction to support
+any kind of arithmetics between `SkyModel` objects, but the number of use
+cases for other operators, except for "+" is very limited and
+can always be achieved by implementing a custom model. I/O and component handling
+of this hierarchical model component structure is intrinsically difficult. For
+this reason we propose to first support and improve the existing `SourceModels` that
+implements an easier to handle flat hierarchy for model components. Support for
+arbitrary model arithmetics can be introduced, if needed, after Gammapy v1.0. We
+propose to remove the `CompoundSourceModel` and reimplement the `+` operator using
+the `SourceModels` class.
+
+.. code-block::
+
+    from gammapy.cube import SourceModel, SourceModels
+
+    component_1 = SourceModel()
+    component_2 = SourceModel()
+
+    total_model = component_1 + component_2
+    assert isinstance(total_model, SourceModels)
+
+    # becomes quivalent to
+
+    total_model = SourceModels([component_1, component_2])
+
+Possibly remove the `SpectralCompoundModel` for consistency.
+
+
+Introduction of model name attributes
+-------------------------------------
+The `SourceModel` class should be improved to support model component names. This
+way the `SourceModels` object can be improved to access model components by name,
+as well as the XML serialization (which supports names) becomes complete.
+The following example should work:
+
+.. code-block::
+
+    from gammapy.cube import SourceModel, SourceModels
+
+    component_1 = SourceModel(name="source_1")
+    component_2 = SourceModel(name="source_2")
+
+    total_model = SourceModels([component_1, component_2])
+
+    total_model["source_1"].parameters["index"].value = 2.3
+
+    # or alternatively??
+
+    total_model.parameters["source_1.index"].value = 2.3
+
+    # delete a model component in place
+    del model["source_2"]
+
+This is also simplifies the parameter access in the fitting back-end, because
+parameter names become unique. E.g. no need for cryptic `par_00X_` parameter names
```

Comment:
Suggest to be more clear on how names are supposed to work. It's a core and important part of the modeling framework.

What if two components or parameters have the same name? Will this give an error on init? Or silently the first match is picked (that's what we do now)?

It's not clear to me if names will always be `modelname.parname` or if more nesting / hierarchy will be supported and used. Can you clarify a bit?